lib/streamlit/DeltaGenerator.py

```diff
@@ -2351,16 +2351,20 @@ def time_input(self, element, label, value=None, key=None):
         return current_value
 
     @_with_element
-    def date_input(self, element, label, value=None, key=None):
+    def date_input(self, element, label, value=None, min_value=datetime.min, max_value=None, key=None):
         """Display a date input widget.
 
         Parameters
         ----------
         label : str
             A short label explaining to the user what this date input is for.
-        value : datetime.date/datetime.datetime
+        value : datetime.date or datetime.datetime
             The value of this widget when it first renders. This will be
             cast to str internally. Defaults to today.
+        min_value : datetime.date or datetime.datetime
+            The minimum selectable date. Defaults to datetime.min.
+        max_value : datetime.date or datetime.datetime
+            The maximum selectable date. Defaults to 2030/12.
```

Comment:
Come to think of it, we should make the default max be `today+10 years`. (It would be even better if the default could be infinity... but not sure that's supported)

`today+10y` is a little less arbitrary than `2030/12`. Also: since `2030/12` is set in BaseWeb's code, if they ever change that we'll have to update our docs to match. Sounds annoying...

Reply:
done